grading/load_config_json.cpp

```diff
@@ -123,37 +130,48 @@ void AddGlobalDefaults(nlohmann::json &whole_config) {
     whole_config["gradeable_message"] = whole_config.value("assignment_message","");
     whole_config.erase("assignment_message");
   }
-  whole_config["gradeable_message"] = whole_config.value("gradeable_message", "");
 
-  whole_config["load_gradeable_message"] = whole_config.value("load_gradeable_message", nlohmann::json::object());
-  whole_config["load_gradeable_message"]["message"] = whole_config["load_gradeable_message"].value("message", "");
-  whole_config["load_gradeable_message"]["first_time_only"] = whole_config["load_gradeable_message"].value("first_time_only", false);
+  json_set_default(whole_config,"gradeable_message",std::string(""));
 
-  whole_config["early_submission_incentive"] = whole_config.value("early_submission_incentive", nlohmann::json::object());
-  whole_config["early_submission_incentive"]["message"] = whole_config["early_submission_incentive"].value("message", "");
-  whole_config["early_submission_incentive"]["minimum_days_early"] = whole_config["early_submission_incentive"].value("minimum_days_early", 0);
-  whole_config["early_submission_incentive"]["minimum_points"] = whole_config["early_submission_incentive"].value("minimum_points", 0);
-  whole_config["early_submission_incentive"]["test_cases"] = whole_config["early_submission_incentive"].value("test_cases", std::vector<int>());
-
-  whole_config["max_submission_size"] = whole_config.value("max_submission_size",MAX_SUBMISSION_SIZE);
+  if (whole_config.find("load_gradeable_message") == whole_config.end()) {
```

Comment:
Could we have here `json_set_default(whole_config, "load_gradeable_message", nlohmann::json::object());`, and so we have just the else branch, a la:

```c++
  json_set_default(whole_config, "load_gradeable_message", nlohmann::json::object());
  json_set_default(whole_config["load_gradeable_message"],"message",std::string(""));
  json_set_default(whole_config["load_gradeable_message"],"first_time_only",false);
```

The doc for [`json::object`](https://nlohmann.github.io/json/api/basic_json/object/).